packages/gas-fee-controller/src/GasFeeController.ts

```diff
@@ -461,18 +410,47 @@ export class GasFeeController extends PollingController<
   async _fetchGasFeeEstimateData(
     options: FetchGasFeeEstimateOptions = {},
   ): Promise<GasFeeState> {
-    const { shouldUpdateState = true } = options;
-    let isEIP1559Compatible;
-    const isLegacyGasAPICompatible =
+    const { shouldUpdateState = true, networkClientId } = options;
+
+    let ethQuery,
+      isEIP1559Compatible,
+      isLegacyGasAPICompatible,
+      decimalChainId: number;
+
+    if (networkClientId !== undefined) {
+      const networkClient = this.messagingSystem.call(
+        'NetworkController:getNetworkClientById',
+        networkClientId,
+      );
+      isLegacyGasAPICompatible = networkClient.configuration.chainId === '0x38';
+
+      decimalChainId = convertHexToDecimal(networkClient.configuration.chainId);
+
+      try {
+        const result = await this.messagingSystem.call(
+          'NetworkController:getEIP1559Compatibility',
+          networkClientId,
+        );
+        isEIP1559Compatible = result || false;
+      } catch {
+        isEIP1559Compatible = false;
+      }
+      // @ts-expect-error TODO: Provider type alignment
+      ethQuery = new EthQuery(networkClient.provider);
+    }
+
+    ethQuery ??= this.ethQuery;
+
+    isLegacyGasAPICompatible ??=
       this.getCurrentNetworkLegacyGasAPICompatibility();
 
-    const decimalChainId = convertHexToDecimal(this.currentChainId);
+    decimalChainId ??= convertHexToDecimal(this.currentChainId);
```

Comment:
can we set these when they are declared on lines 415 to 418 as opposed to checking if they were set by networkClientId flow and using nullish coalesce?

Reply:
I guess the benefit to the way I have it now is that it avoids additional unnecessary calls since if the values to the left of the nullish assignment are defined the right side is never called.

Reply:
fair enough